psycopg/utils.c

```diff
@@ -50,8 +50,13 @@ psycopg_escape_string(connectionObject *conn, const char *from, Py_ssize_t len,
     Py_ssize_t ql;
     int eq = (conn && (conn->equote)) ? 1 : 0;
 
-    if (len == 0)
+    if (len == 0) {
         len = strlen(from);
+    } else if (strcspn(from, "\0") != len) {
```

Comment:
Derp, how ironic that I missed that in a fix for a nul-terminator bug. I've adapted the PR to use strchr instead.